### ISSUE TITLE
Some trivial XCB bugs

### ISSUE THREAD

**Issue by jcowgill** [LaurentGomila/SFML#786]
Found with clang static code analyzer:

https://github.com/LaurentGomila/SFML/blob/master/src/SFML/Window/Unix/InputImpl.cpp#L205
```
src/SFML/Window/Unix/InputImpl.cpp:205:31: warning: Value stored to 'result' is never read
        case Mouse::Left:     result = pointer->mask & XCB_BUTTON_MASK_1;
                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SFML/Window/Unix/InputImpl.cpp:206:31: warning: Value stored to 'result' is never read
        case Mouse::Right:    result = pointer->mask & XCB_BUTTON_MASK_3;
                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/SFML/Window/Unix/InputImpl.cpp:207:31: warning: Value stored to 'result' is never read
        case Mouse::Middle:   result = pointer->mask & XCB_BUTTON_MASK_2;
                              ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

https://github.com/LaurentGomila/SFML/blob/master/src/SFML/Window/Unix/WindowImplX11.cpp#L538
```
src/SFML/Window/Unix/WindowImplX11.cpp:538:73: warning: Use of memory after it is freed
        err() << "Failed to set the window's icon: Error code " << (int)errptr->error_code << std::endl;
                                                                        ^~~~~~~~~~~~~~~~~~
```

**Comment by LaurentGomila** [comment]
The first one is less obvious: the "break" statements are missing. It would be much better like this:

```cpp
    // Get pointer mask
    xcb_query_pointer_reply_t* pointer = xcb_query_pointer_reply(connection, xcb_query_pointer(connection, XDefaultRootWindow(display)), NULL);
    uint16_t mask = pointer->mask;
    free(pointer);

    // Close the connection with the X server
    CloseDisplay(display);

    switch (button)
    {
        case Mouse::Left:     return pointer->mask & XCB_BUTTON_MASK_1;
        case Mouse::Right:    return pointer->mask & XCB_BUTTON_MASK_3;
        case Mouse::Middle:   return pointer->mask & XCB_BUTTON_MASK_2;
        case Mouse::XButton1: return false; // not supported by X
        case Mouse::XButton2: return false; // not supported by X
        default:              return false;
    }
```

**Comment by kimci86** [comment]
@LaurentGomila
I guess you meant to return `mask & ...` instead of `pointer->mask & ...`

**Comment by LaurentGomila** [comment]
Oops :grin:

**Comment by zwookie** [comment]
I'm not sure if this is the right place to put this, but I ran into an XCB related issue trying to build my project with the latest master branch for SFML. Here's the output I got using CodeBlocks with static release libraries.

`/usr/bin/ld: /dev/SFML-build/lib/libsfml-window-s.a(WindowImplX11.cpp.o)||undefined reference to symbol 'XSetEventQueueOwner'|`

I'm using Ubuntu 14.04. I just installed all the libxcb-xxxx-dev packages that seemed to be required for CMake.
